### PR TITLE
feat(datestore): naive version implementation, add records timestamp

### PR DESCRIPTION
**Pull request by xuchuan**
## Modules
- [ ] UI
- [X] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [X] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java [hunk 1]
```diff
@@ -264,19 +293,45 @@ public void update(TableSchemaDesc schema, List<Map<String, Object>> records) {
...
+                deletedFlag = false;
+            }
+            var versions = this.recordMap.computeIfAbsent(key, k -> new ArrayList<>());
+            if (deletedFlag) {
+                if (versions.isEmpty() || !versions.get(versions.size() - 1).isDeleted()) {
+                    versions.add(MemoryRecord.builder()
+                            .timestamp(timestamp)
+                            .deleted(true)
+                            .build());
+                }
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java:316]: Can this block be removed? 
It seems that it's just a special condition of the logic below.

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/DataStore.java [hunk 2]
```diff
@@ -324,27 +353,36 @@ class TableRecords {
...
                     return x.compareTo(y);
                 }).record.getKey();
-                var record = new HashMap<String, Object>();
+                Map<String, Object> record = null;
                 for (var r : records) {
                     if (r.record.getKey().equals(lastKey)) {
-                        record.putAll(
-                                DataStore.encodeRecord(r.meta.columnTypeMap, r.record.values, req.isRawResult()));
+                        if (r.record.isDeleted()) {
+                            record = null;
```
[the remark this answers is not in the document]
Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/DataStore.java:365]: "scan" returns a snapshot based on the timestamp parameter. Syncing data between instances with history is another topic. We may add this feature in the future.

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java (hunk 1 above)
Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/impl/MemoryTableImpl.java:316]: Their conditions are different: "!versions.get(versions.size() - 1).isDeleted()" vs "!newRecord.isEmpty()", so it is better to put them in two clauses.

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**